### ISSUE TITLE
Wordmove not copying file in wp-content

### ISSUE THREAD

**Issue by defaria** [welaika/wordmove#214]
I use a plugin called Login Logo, which looks for an image in the wp-content folder called login-logo.png, and then set's that image as the login logo. On 3 different websites wordmove failed to copy that image over. The only excluded settings I have for wp-content is the one that comes by default with the movefile.

"wp-content/*.sql"

Has anyone else noticed issues copying files from wp-content?

Just to be clear... other folders inside wp-content are copied just fine. But the .png image is the only actual file I have in there.

**Comment by pioneerskies** [comment]
It could be useful in these cases to report the command you are issuing to Wordmove.

Anyway: I think this is related to #163 and is a design issue. `-w` flag pushes the core excluding `wp-content/*`, other flags keep care of the directories inside `wp-content`. This way we don't care about files inside `wp-content`.

Your case is someway exotic IMHO, since your plugin searches for something in a non standard path and in a non standard way (save here an image and I will search for it is a really weak approach from my POW).

I'd like anyway to continue the discussion together with @luisbraga and poll you a bit about what should `Wordmove` be doing different from the actual behaviour.

**Comment by defaria** [comment]
Thanks @pioneerskies. I was simply doing a pull --all , to get a copy of the site down to local.

**Comment by Mark-William-Schumacher** [comment]
I'm also experiencing a missed upload when i push with 
"wordmove push -e staging --all"

The newly added files in /wp-content/webfonts/ are all missing from the upload. 

The issue was resolved when i put the webfonts folder into /wp-content/uploads/webfonts/ instead.

**Comment by filipvanreeth** [comment]
Hi, i have the same problem. The wp-content folder with it's content is not getting pushed to the development version in my case. I use the command wordmove push -e staging --all and i'm using the default Movefile file.

Anything I'm doing wrong?

**Comment by vojtabiberle** [comment]
Hi. I have same problem with nextgen-gallery. Files in wp-content/gallery are not synced.

**Comment by mukkoo** [comment]
Wordmove doesn't sync all directories inside wp-content/ but only languages, themes, plugins and mu-plugins. This can be a feature?

**Comment by filipvanreeth** [comment]
Thanks. That would be nice

**Comment by alexanderkladov** [comment]
Hi,

I was trying to figure out why my `wp-content/index.php` wasn't syncing over and came across this thread. I have that file for security reasons (a.k.a. silence is golden). And after reading this thread I've realised that Wordmove is adding `wp-content/*` to my Excludes, even though I didn't add that myself.

Why was this removed from the v2.2.0 milestone and how can I prevent that from happening?

Cheers,
Alex

**Comment by ihorvorotnov** [comment]
> Wordmove doesn't sync all directories inside wp-content/ but only languages, themes, plugins and mu-plugins. This can be a feature?

I can understand this design decision. However, it misses some 'core' files, for example, drop-ins:

path/file | description
--- | ---
`wp-content/object-cache.php` | external object cache class
`wp-content/advanced-cache.php` | alternative cache class
`wp-content/db.php` | custom db class
`wp-content/db-error.php` | custom db error message
`wp-content/install.php` | custom installation routine
`wp-content/maintenance.php` | custom maintenance mode message
`wp-content/sunrise.php` | multisite preloader
`wp-content/blog-deleted.php` | custom message if a multisite blog was deleted
`wp-content/blog-inactive.php` | custom message if a multisite blog was set inactive
`wp-content/blog-suspended.php` | custom message if a multisite blog was suspended
`wp-content/$locale.php` | is loaded for a specific locale in use

Also, the `wp-content` dir itself is considered as a directory to keep everything that's 'user generated content' or not plugin/theme. That is the base purpose of the dir. Custom dirs within `wp-content` are a common practice, especially in custom development setups with Composer and modern approaches.

Basically, I'd expect to find anything inside the `wp-content` dir. That's what it is for.

**Comment by pioneerskies** [comment]
@alexanderkladov about this

> Also, while on the topic of excludes: how can I ignore a file ONLY in root wp directory? I've tried ./filename & /filename, but neither one worked..

It was hopefully fixed in https://github.com/welaika/wordmove/commit/164206589aa680a1086b3ff9f7af134acbe7dacb

**Comment by ihorvorotnov** [comment]
@pioneerskies I'm new to Wordmove (like, couple days new) so I don't have any preferred way yet. Based on logic and previous experience with other tools/techs, I'd be happy with the default option to move entire wp-content and exclude only dirs/files that are specifically listed in the config.

**Comment by alexanderkladov** [comment]
@pioneerskies I agree with @ihorvorotnov, push everything in `wp-config` or even root website folder, unless specifically excluded in `movefile.yml`. Just like what Git would do (i.e. if it's not in `.gitignore`, it's pushed to the repo).

**Comment by kauaicreative** [comment]
`wordmove pull --content` could work.  
Google's new plugin/theme stores style data in `/wp-content/assets/css/front-end-compiled.css`
https://wordpress.org/plugins/material-design/

**Comment by kauaicreative** [comment]
Being able to push pull a custom directory in wp-content would be ideal `wordmove pull --custom=cache
